sfbrigade/datasci-earthquake
# Data handling

# Description

Creates data handlers for soft story properties and liquefaction zones.

Addresses but does not resolve #113

## Type of changes

## Testing

## How to test

1. Start docker with
2. `docker compose up`
3. Run docker exec -it datasci-earthquake-backend-1 python backend/database/init_db.py to initialize the tables.
4. Run docker exec -it datasci-earthquake-backend-1 python backend/etl/liquefaction_data_handler.py to populate the table.
5. To verify that the table is populated, run docker exec -it my_postgis_db psql -U postgres -d qsdatabase , which will open a psql terminal; then enter the following queries and expect the following results:

```
select count(*) from liquefaction_zones;
```
```
 count 
-------
     2
(1 row)
```
```
select liq from liquefaction_zones;
```
```
 liq 
-----
 H
 VH
(2 rows)
```
```
select shape_length from liquefaction_zones;
```
```
 shape_length  
---------------
 17664251.5482
 25527219.0905
(2 rows)
```
```
select shape_area from liquefaction_zones;
```
```
  shape_area   
---------------
 9247150010.84
 4358670967.78
(2 rows)
```
```
select update_timestamp from liquefaction_zones;
```
```
       update_timestamp        
-------------------------------
 2024-12-08 18:48:22.120779+00
 2024-12-08 18:48:22.120779+00
(2 rows)
```
